### ISSUE TITLE
bug(mac): Keyman menu absent after installing on Catalina or Big Sur

### ISSUE THREAD

**Issue by MakaraSok** [keymanapp/keyman#9060]
### Describe the bug

Keyman 16.0.139 (stable version) got installed successfully on Catalina 10.15.7, but the Keyman menu does not show up on the taskbar menu for Keyman where the Keyman logo is at.

### Expected behavior

The Keyman menu items, such as: `Keyboards`, `Configuration...`, `On-Screen Keyboard` and `About`, should be shown after clicking on the Keyman icon on the taskbar.

This doesn't get shown on Catalina.
<img width="254" alt="image" src="https://github.com/keymanapp/keyman/assets/28331388/5725fa8c-a632-4c05-acc7-a3c8422c28f4">

### Keyman apps

- [ ] Keyman for Android
- [ ] Keyman for iPhone and iPad
- [ ] Keyman for Linux
- [X] Keyman for macOS
- [ ] Keyman for Windows
- [ ] Keyman Developer
- [ ] KeymanWeb
- [ ] Other - give details at bottom of form

### Keyman version

16.0.139

### Operating system

Catalina 10.15.7

### Device

MacBook Pro, Early 2013

### Additional context

![image](https://github.com/keymanapp/keyman/assets/28331388/89a98bb8-0e92-421f-bdf3-62a5f07d644f)

**Comment by MakaraSok** [comment]
related https://community.software.sil.org/t/configuration-window-on-mac/7930

**Comment by bharanidharanj** [comment]
@MakaraSok  Tested this issue in macOS Catalina 10.15.3 along with Keyman 16.0.141 stable build and here is my observation: 1. After installing Keyman build, I could only see the keyman icon on the status bar. I noticed that the Keyman menu was missing on the taskbar. 2. I am able to reproduce this issue.

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/8a15ef84-4c52-41e5-8ea0-641f747f277e">

**Comment by bharanidharanj** [comment]
@MakaraSok Retested with Keyman 17.0.178 build in macOS Catalina and I am able to see the same issue.

**Comment by MakaraSok** [comment]
Thank you @bharanidharanj for the confirmation. It looks like Catalina is not happy with Keyman, or another way around.

The system requirements of Keyman 16.0 is from 10.10 up per KeymanHelp documentation found at https://help.keyman.com/products/mac/current-version/about/requirements.

**Comment by sgschantz** [comment]
may be fixed when #2542 is addressed, but steps to reproduce would be helpful

**Comment by shiami** [comment]
> @MakaraSok Retested with Keyman 17.0.178 build in macOS Catalina and I am able to see the same issue.

Keyman 16.0.147 in macOS Catalina 10.15.7, the same issue.

**Comment by mcdurdin** [comment]
I closed two issues as duplicates of this. Note additional comments from:
* #10217
* #11180

**Comment by Nnyny** [comment]
see also: https://community.software.sil.org/t/no-keyman-menu-on-mac/8991

**Comment by sgschantz** [comment]
This is actually occuring due to a crash at startup with any os prior to 12.0, (Monterey)

From crash report, running on macOS Big Sur (11.7.10):

```
Termination Reason:    DYLD, [0x4] Symbol missing

Application Specific Information:
dyld: launch, loading dependent libraries

Dyld Error Message:
  Symbol not found: __ZTTNSt3__114basic_ifstreamIcNS_11char_traitsIcEEEE
  Referenced from: /Users/USER/Library/Input Methods/Keyman.app/Contents/MacOS/../Frameworks/KeymanEngine4Mac.framework/Versions/A/KeymanEngine4Mac
  Expected in: /usr/lib/libc++.1.dylib
 in /Users/USER/Library/Input Methods/Keyman.app/Contents/MacOS/../Frameworks/KeymanEngine4Mac.framework/Versions/A/KeymanEngine
```

**Comment by sgschantz** [comment]
Fixed by #12223
Keyman Core was being built for a minimum macOS version of 12, so any OS version less than 12, such as Catalina or Big Sur, had trouble dynamically loading KeymanEngine which depends on Keyman Core.
This does not explain why Keyman 16 would encounter this issue (as described above) -- only 17 and later of Keyman for Mac use Keyman core. However, I tested with Keyman 16 on macOS Big Sur and was unable to reproduce the issue.
